### PR TITLE
more doxygen changes

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #191 

Next batch of code files converted to doxygen.

### CODE REVIEW

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
@edwardhartnett, if you check the syntax of pmat.f90, I can merge this next.

File: sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90 [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
-! the form of an optional "failure flag", FF, which is normally returned
-! as .FALSE., but is returned as .TRUE. when inversion fails.
-! In Sep 2012, these routines were collected together into pmat.f90 so
-! that all the main matrix routines could be in the same library, pmat.a.
-! 
-! DIRECT DEPENDENCIES:
-! Modules: pkind, pietc
-!
-!=============================================================================
+!> @date
```
[the remark this answers is not in the document]
Comment (edwardhartnett) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90:1]: OK, stand by, lots of meetings today!

[the remark this answers is not in the document]
Comment (edwardhartnett) [reply sorc/grid_tools.fd/regional_esg_grid.fd/pmat.f90]: OK, fixed.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
Everything looks correct.  Will merge.

**Comment by GeorgeGayno-NOAA** [comment]
@edwardhartnett Changes look good.  But there is a warning about your branch being out of date with the base branch.  Try another merge from develop.

**Pull request merged**